### PR TITLE
refactor(required_fields): change required fields for billing address

### PR DESCRIPTION
**Pull request by Narayanbhat166**
## Type of Change

- [x] Refactoring

## Description

This PR changes the dynamic fields to collect `card_holder_name` as `payment_method_data.billing.first_name` and `payment_method_data.billing.last_name`. Backwards compatibility has been provided, so if `card_holder_name` is passed, then it will override the `payment_method_data.billing.first_name` and `payment_method_data.billing.last_name` passed in billing details.

All the required fields for billing details are changed to be collected as `payment_method_data.billing`

### Additional Changes

- [x] This PR modifies the API contract

The billing details will now be collected in payment_method_billing.

## Motivation and Context

Standardise the way in which to collect billing details.

## How did you test it?

- Create a payment via sdk, check for connectors where `card_holder_name` is mandatory.
image

- Check the /confirm call and confirm if card holder name is passed in billing details along with other billing fields.
image

- Check whether required fields are rendered correctly by passing billing details in creating the payment and not passing billing details in creating the payment. When billing details are passed in payment create, then required fields should not be rendered. But when they are passed, required fields should be rendered in the sdk.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

**Review by ArjunKarthik — DISMISSED** [review]

File: crates/router/src/configs/defaults.rs [hunk 1]
```diff
@@ -2048,20 +2138,20 @@ impl Default for super::settings::RequiredFields {
...
+                                                    required_field: "payment_method_data.billing.address.first_name".to_string(),
                                                     display_name: "card_holder_name".to_string(),
-                                                    field_type: enums::FieldType::UserBillingName,
+                                                    field_type: enums::FieldType::UserFullName,
                                                     value: None,
                                                 }
                                             ),
                                             (
-                                                "billing.address.last_name".to_string(),
+                                                "payment_method_data.billing.address.last_name".to_string(),
```
Comment (prasunna09) [inline crates/router/src/configs/defaults.rs:2150]: here, we are using `payment_method_data` 's last name but for fields line1, city, we are using `billing.address.line1`, `billing.address.city`. Is there any reason we are using two different objects?

**Review by prasunna09 — CHANGES_REQUESTED** [review]

File: crates/router/src/configs/defaults.rs (hunk 1 above)
Comment (Narayanbhat166) [reply crates/router/src/configs/defaults.rs]: I have changed the key to refer the value from `billing.address.first_name` and `last_name`

**Review by prasunna09 — DISMISSED** [review]

File: crates/router/src/configs/defaults.rs [hunk 2]
```diff
@@ -742,7 +760,7 @@ impl Default for super::settings::RequiredFields {
                                             (
                                                 "billing.address.first_name".to_string(),
                                                 RequiredFieldInfo {
-                                                    required_field: "billing.address.first_name".to_string(),
+                                                    required_field: "payment_method_data.billing.address.first_name".to_string(),
                                                     display_name: "billing_first_name".to_string(),
                                                     field_type: enums::FieldType::UserBillingName,
```
Comment (ArjunKarthik) [inline crates/router/src/configs/defaults.rs:765]: this should be UserFullName

File: crates/router/src/configs/defaults.rs [hunk 3]
```diff
@@ -2733,16 +2787,16 @@ impl Default for super::settings::RequiredFields {
...
                                                     value: None,
                                                 }
                                             ),
                                             (
                                                 "billing.address.last_name".to_string(),
                                                 RequiredFieldInfo {
-                                                    required_field: "billing.address.last_name".to_string(),
+                                                    required_field: "payment_method_data.billing.address.last_name".to_string(),
                                                     display_name: "billing_last_name".to_string(),
                                                     field_type: enums::FieldType::UserBillingName,
```
Comment (ArjunKarthik) [inline crates/router/src/configs/defaults.rs:2801]: here too

File: crates/router/src/configs/defaults.rs [hunk 4]
```diff
@@ -2809,7 +2863,7 @@ impl Default for super::settings::RequiredFields {
                                             (
                                                 "billing.address.first_name".to_string(),
                                                 RequiredFieldInfo {
-                                                    required_field: "billing.address.first_name".to_string(),
+                                                    required_field: "payment_method_data.billing.address.first_name".to_string(),
                                                     display_name: "billing_first_name".to_string(),
                                                     field_type: enums::FieldType::UserBillingName,
```
Comment (ArjunKarthik) [inline crates/router/src/configs/defaults.rs:2868]: +1

File: crates/router/src/configs/defaults.rs [hunk 5]
```diff
@@ -3398,16 +3434,16 @@ impl Default for super::settings::RequiredFields {
...
                                                     value: None,
                                                 }
                                             ),
                                             (
                                                 "billing.address.last_name".to_string(),
                                                 RequiredFieldInfo {
-                                                    required_field: "billing.address.last_name".to_string(),
+                                                    required_field: "payment_method_data.billing.address.last_name".to_string(),
                                                     display_name: "billing_last_name".to_string(),
                                                     field_type: enums::FieldType::UserBillingName,
```
Comment (ArjunKarthik) [inline crates/router/src/configs/defaults.rs:3448]: +1

File: crates/router/src/configs/defaults.rs [hunk 6]
```diff
@@ -3558,16 +3594,16 @@ impl Default for super::settings::RequiredFields {
...
```
Comment (ArjunKarthik) [inline crates/router/src/configs/defaults.rs:3608]: +1

File: crates/router/src/configs/defaults.rs (hunk 2 above)
Comment (Narayanbhat166) [reply crates/router/src/configs/defaults.rs:765]: Should I also change this to `card_holder_name`?

**Review by prasunna09 — APPROVED** [review]

**Pull request merged**